Automattic/newspack-plugin
# Push Notifications

### All Submissions:






### Changes proposed in this Pull Request:

Implements basic support for AMP-compatible Push Notifications using [One Signal](https://onesignal.com/). This PR still has some rough edges but is ready for review/discussion.

Video demo of subscribing: https://cloudup.com/cyyfKRs0tG7

### How to test the changes in this Pull Request:

1. Navigate to Newspack Engagement Wizard, click on "Push" tab.
2. Click toggle to install One Signal plugin.
3. Go to https://onesignal.com/ create a free account, then create an App for "Web Push."
4. In Settings->Keys/IDs find your APP ID and REST API KEY. Copy paste these into the corresponding fields in the engagement wizard.
5. Edit a widget area. Add the Newspack Push Notifications widget to a an area. 
6. Try testing in an AMP request. You should see a Subscribe button. Clicking the button opens a popup browser window, and in the window you will be prompted to subscribe. 
7. After subscribing you should see the number of subscribers increment in the One Signal dashboard. 
8. If you don't, you may be blocking notifications. On advice of @adekbadek I usually would create a new Chrome profile (click your Avatar in the upper right, choose ADD+), and subscribe with that user. 
9. After successfully subscribing, try creating and publishing a post and you should receive a push notification.

 ### Open Questions

1. For now it appears the pop-up window is required. I'll be opening an AMP issue seeking improvement but for now it seems we're stuck with it. How should we style to make it a more palatable experience? cc: @thomasguillot 
2. How can we make the widget UI better?
3. Should we add a block as well? If so, it would presumably live in Newspack Blocks.
4. Should we guide users through acquiring and configuring their One Signal account? There is some information about doing this in the One Signal plugin — we could just send new users over there. 
5. I believe the implementation will not currently work in non-AMP. This should be addressed.

### Other information:
